src/strategies/offer_forwarder/abstract/Forwarder.sol

```diff
@@ -308,13 +308,13 @@ abstract contract Forwarder is IForwarder, MangroveOffer {
     // NB if several offers of `this` contract have failed during the market order, the balance of this contract on Mangrove will contain cumulated free provision
 
     // computing an under approximation of returned provision because of this offer's failure
-    (MgvStructs.GlobalPacked global, MgvStructs.LocalPacked local) = MGV.config(order.outbound_tkn, order.inbound_tkn);
-    uint provision =
-      10 ** 9 * order.offerDetail.gasprice() * (order.offerDetail.gasreq() + order.offerDetail.offer_gasbase());
+    uint gasreq = order.offerDetail.gasreq();
+    uint provision = 10 ** 9 * order.offerDetail.gasprice() * (gasreq + order.offerDetail.offer_gasbase());
 
     // gasUsed estimate to complete posthook and penalize this offer is ~1750 (empirical estimate)
-    uint approxBounty =
-      (order.offerDetail.gasreq() - (gasleft() - GAS_APPROX) + local.offer_gasbase()) * global.gasprice() * 10 ** 9;
+    uint gasprice = order.global.gasprice() * 10 ** 9;
+    uint approxGasConsumption = gasreq + GAS_APPROX + order.local.offer_gasbase();
+    uint approxBounty = (approxGasConsumption - gasleft()) * gasprice;
```

Comment:
if this reverts for underflow are we good? I think so because everything works as if offer logic had used the whole gasreq (and therefore there is nothing to claim from provisions).

Reply:
x-posting Lasse's response from Slack:

1. the underflow risk was already there before, right?
2. I think it warrants a comment on why this should not underflow - is gasleft() < gasreq?
3. Could we just change approxBounty to int ? Or will that enable other over/underflow problems in the comparison? I am not sure how (int)approxBounty >= (uint)provision works in solidity :smile: